XamarinCommunityToolkit/Behaviors/AttachedNumericValidationBehavior.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xamarin.Forms;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    public static class AttachedNumericValidationBehavior
+    {
+        public static readonly BindableProperty AttachBehaviorProperty =
+            BindableProperty.CreateAttached("AttachBehavior", typeof(bool), typeof(AttachedNumericValidationBehavior), false, propertyChanged: OnAttachBehaviorChanged);
+
+        public static bool GetAttachBehavior(BindableObject view)
+        {
+            return (bool)view.GetValue(AttachBehaviorProperty);
+        }
+
+        public static void SetAttachBehavior(BindableObject view, bool value)
+        {
+            view.SetValue(AttachBehaviorProperty, value);
+        }
+
+        static void OnAttachBehaviorChanged(BindableObject view, object oldValue, object newValue)
+        {
+            var entry = view as Entry;
+
+            if (entry == null)
+            {
+                return;
+            }
+
+            bool attachBehavior = (bool)newValue;
+
+            if (attachBehavior)
+            {
+                entry.TextChanged += OnEntryTextChanged;
+            }
+            else
+            {
+                entry.TextChanged -= OnEntryTextChanged;
+            }
+        }
+
+        static void OnEntryTextChanged(object sender, TextChangedEventArgs args)
+        {
+            double result;
+            bool isValid = double.TryParse(args.NewTextValue, out result);
+            ((Entry)sender).TextColor = isValid ? Color.Default : Color.Red;
+        }
```

Comment:
```csharp
            ((Entry)sender).TextColor = double.TryParse(args.NewTextValue, out double result) 
                ? Color.Default 
                : Color.Red;
```

Reply:
Also, I believe colors should be customizable

Reply:
I agree.  This is an old class from years ago.  Any suggestions?  Include another BindableProperty for Color?